### PR TITLE
[ros2topic] Add option to echo serialized messages

### PR DESCRIPTION
**Pull request by jacobperron**
I found this feature useful when doing some debugging.

~~Depends on https://github.com/ros2/rosidl_runtime_py/pull/10~~

### CODE REVIEW

File: ros2topic/test/test_echo_pub.py [hunk 1]
```diff
@@ -245,3 +245,43 @@ def publish_message():
...
+        assert publisher
+
+        def publish_message():
+            publisher.publish(String(data='hello'))
+
+        publish_timer = self.node.create_timer(0.5, publish_message)
+
+        try:
+            command_action = ExecuteProcess(
+                cmd=(['ros2', 'topic', 'echo', '--raw', topic, 'std_msgs/msg/String']),
```
Comment (hidmic) [inline ros2topic/test/test_echo_pub.py:262]: @jacobperron no need for the extra parenthesis.

File: ros2topic/test/test_echo_pub.py [hunk 2]
```diff
@@ -245,3 +245,43 @@ def publish_message():
...
+            with launch_testing.tools.launch_process(
+                launch_service, command_action, proc_info, proc_output,
+                output_filter=launch_testing_ros.tools.basic_output_filter(
+                    filtered_rmw_implementation=get_rmw_implementation_identifier()
+                )
+            ) as command:
+                # The future won't complete - we will hit the timeout
+                self.executor.spin_until_future_complete(
+                    rclpy.task.Future(), timeout_sec=5
+                )
```
Comment (hidmic) [inline ros2topic/test/test_echo_pub.py:277]: @jacobperron this is a bit flaky -- why not waiting for output from the `ros2 topic echo` command?

Comment (jacobperron) [reply ros2topic/test/test_echo_pub.py:341]: I think this spin is necessary for the test publisher to work (it is triggered by a timer registered above). This is similar logic to the existing test in this file:

https://github.com/ros2/ros2cli/blob/95bfb056df795b6eef8bb73b165db31fcf2d867a/ros2topic/test/test_echo_pub.py#L276-L278

I guess we could additionally wait for output after this spin, which should be more robust. I'll give it a try.

File: ros2topic/test/test_echo_pub.py (hunk 1 above)
Comment (jacobperron) [reply ros2topic/test/test_echo_pub.py]: removed in 72d5e00

File: ros2topic/test/test_echo_pub.py (hunk 2 above)
Comment (jacobperron) [reply ros2topic/test/test_echo_pub.py:342]: See 4472a9e

**Comment by jacobperron** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11475/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6680/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9411/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11368/)

**Comment by jacobperron** [comment]
@ros-pull-request-builder retest this please

**Comment by jacobperron** [comment]
After rebase:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11591/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6732/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9471/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11553/)

**Pull request merged**